src/multitenant/mulittenant.service.ts

```diff
@@ -0,0 +1,199 @@
+            token = result.token;
+        } catch (e) {
+            // Handle already registered case
+            if (e.details && e.details.match(/Wallet with name \w+ already exists/g)) {
+                Logger.log('Wallet already exists, fetching data from storage');
+                [walletId, token] = await this.getTokenAndWalletId(body.walletName, body.walletKey);
+            } else {
+                throw new ProtocolException(ProtocolErrorCode.INTERNAL_SERVER_ERROR, 'Error creating wallet on multitenant', e);
+            }
+        }
+
+        // Add to cache
+        const ttl = (body.ttl === null || body.ttl === undefined) ? this.DEFAULT_TTL_SECONDS : body.ttl;
+        let invitation = null;
+        await this.addWalletToCache(body.walletName, body.walletKey, walletId, token, ttl);
+
+        // Set remove job
+        await this.setRemoveJob(walletId, body.walletKey, ttl);
+
+        // Handle auto connect
+        if (body.autoConnect) {
+            invitation = await this.callCreateConnection(token);
+        }
+
+        // Only return token in local/test envs
+        if (process.env.NODE_ENV !== Constants.LOCAL) {
+            token = null;
+        }
+
+        return {
+            success: true,
+            invitation,
+            token,
+        };
+    }
+
+    /**
+     * Remove or unregister a wallet from multitenant
+     * TODO handle case where wallet isn't in cache
+     */
+    public async removeWallet(walletName: string, walletKey: string): Promise<any> {
+        const agent: any = await this.cache.get(walletName);
+        if (!agent) {
+            Logger.error(`Attempted to remove wallet ${walletName} but it wasn't in cache`);
+            throw new ProtocolException(ProtocolErrorCode.INTERNAL_SERVER_ERROR, 'Error removing wallet on multitenant');
+        }
+        return await this.callRemoveWallet(agent.walletId, walletKey);
+    }
+
+    /**
+     * Saves wallet info to cache for easy future access
+     */
+    private async addWalletToCache(walletName: string, walletKey: string, walletId: string, token: string, ttl: number): Promise<void> {
+        // Generally we want the cache to last 1 second longer than the agent, except when set to an "infinite" value like 0 or -1
+        const cacheTtl = (ttl < 0) ? ttl : ttl + 1;
+        Logger.info(`record cache limit set to: ${cacheTtl}`);
+        await this.cache.set(
+            walletName,
+            {
+                adminApiKey: process.env.MULTITENANT_API_KEY,
+                token,
+                walletId,
+                walletKey,
+                ttl,
+                multitenant: true
+            },
+            {
+                ttl: cacheTtl
+            }
+        );
+    }
+
+    /**
+     * ttl = time to live is expected to be in seconds (which we convert to milliseconds).  if 0, then live in eternity
+     */
+    private setRemoveJob(walletId: string, walletKey: string, ttl: number): void {
```

Comment:
Can we comment what the purpose is?  I see in create wallet we call this method and remove wallet, which seems confusing.